### PR TITLE
Lookup benchmark

### PR DESCRIPTION
**Pull request by bright-starry-sky**
Lookup benchmark , test result as below :

**System info :** 

- CPU : Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
- MemTotal : 8G

**Total vertices : 1000000**

- PreciseScan_10000 : Precise scan , return 10000 vertices
- PreciseScan_100   : Precise scan , return 100 vertices
- PreciseScan_10    : Precise scan , return 10 vertices
- FilterScan_10     : Full index scan, return 1 vertex

```
====================================================
StorageLookupBenchmark.cpprelative  time/iter  iters/s
====================================================
PreciseScan_10000                   577.73ms     1.73
PreciseScan_100                    594.96ms     1.68
PreciseScan_10                      469.44ms     2.13
FilterScan_10                          1.57s  636.61m
=====================================================
```

### CODE REVIEW

**Comment by bright-starry-sky** [comment]
**System Info** 

- CPU : Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
- MemTotal : 8G

**Test result**

- Total vertices : 1000000
- 
- - PreciseScan_10000 : Precise scan , return 10000 vertices
- - PreciseScan_100   : Precise scan , return 100 vertices
- - PreciseScan_10    : Precise scan , return 10 vertices
- - FilterScan_10     : Full index scan, return 1 vertex


```
=======================================================
src/storage/test/StorageLookupBenchmark.cpprelative  time/iter  iters/s 

=======================================================
PreciseScan_10000                                          31.81ms     4.12
PreciseScan_100                                            33.98ms     4.63
PreciseScan_10                                             28.96ms     4.52
FilterScan_10                                              260.16ms    2.82
=======================================================
```

File: src/storage/test/StorageLookupBenchmark.cpp [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+ * CPU : Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
+ * MemTotal : 8G
+ * Total vertices : 1000000
+ * PreciseScan_10000 : Precise scan , return 10000 vertices
+ * PreciseScan_100   : Precise scan , return 100 vertices
+ * PreciseScan_10    : Precise scan , return 10 vertices
+ * FilterScan_10     : Full index scan, return 1 vertex
+ **/
+
+/**
```
Comment (dangleptr) [inline src/storage/test/StorageLookupBenchmark.cpp]: What's the index and expression?  
What's the part number?

Comment (bright-starry-sky) [reply src/storage/test/StorageLookupBenchmark.cpp]: > What's the index and expression?
> What's the part number?

- Part number : 
1 part

- tag、index structure as below : 
```
tag (col0 string, col1 string, col2 string)
index on tag(col0, col1, col2)
```

- expression : 

```
PreciseScan : col0 == xxxx
FilterScan : col1 == xxxx
```
The FilterScan is not work now. I need to improve this benchmark at later.

**Comment by bright-starry-sky** [comment]
To improve the benchmark.
Newly test result as below. Performance bottleneck in encode and decode when index full scan.

- CPU : Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
- MemTotal : 8G
- Total vertices : 1000000
- Total partition number : 1
- tag (col_0 int, col_1 int, col_2 int)
- index on tag (col_0, col_1, col_2)
- PreciseScan_10000 : Precise scan , return 10000 vertices , where clause col_1 == 1
- PreciseScan_100   : Precise scan , return 100 vertices , where clause col_1 == 1
- PreciseScan_10    : Precise scan , return 10 vertices , where clause col_1 == 1
- FilterScan_10     : Full index scan, return 1 vertex , where clause col_1 < 1

```
=====================================================
 src/storage/test/StorageLookupBenchmark.cpprelative  time/iter  iters/s
 =====================================================
 PreciseScan_10000                                          9.65ms   103.65
 PreciseScan_100                                            5.23ms   191.03
 PreciseScan_10                                             4.14ms   241.64
 FilterScan_10                                              1.47s  679.35m
 =====================================================
```

File: src/storage/test/StorageLookupBenchmark.cpp (hunk 1 above)
Comment (dangleptr) [reply src/storage/test/StorageLookupBenchmark.cpp]: Fine.  Could you update the comments

Comment (bright-starry-sky) [reply src/storage/test/StorageLookupBenchmark.cpp]: > Fine. Could you update the comments

Sure, comments updated.

**Pull request merged**